support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -335,25 +335,58 @@ export function Checkout({ geoId, appConfig }: Props) {
 		return <div>Rate plan not found</div>;
 	}
 
-	/** Get and validate price */
-	let price: number | undefined;
+	/**
+	 * Get and validate the amount
+	 *
+	 * For products the amount is based on
+	 * - the product price in the catalog
+	 * - any promotions applied
+	 * - any contributions made
+	 */
+
+	/**
+	 * - `originalAmount` the amount pre any discounts or contributions
+	 * - `discountredAmount` the amount with a discountApplied
+	 * - `finalAmount` is the amount a person will pay
```

Comment:
When would the final amount be different to the discounted amount? I could see this happening with supporter plus by adding a contribution, but it doesn't seem like a useful use case

Reply:
The case you explain above is 1 case. It's also to differentiate between the two. If discountedAmount exists, you know a discount has been applied.

I was hoping the naming would capture a universal flow of how a price is calculated and can be used in all cases. 


e.g. `original` - `discounted` + `contribution` = `final` and then we can use those values in different places.

Reply:
This does make sense. But could we pass around the amount of the discount instead? i.e. the `discounted` in your equation

Reply:
Do we mean removing `final` and just use `discounted`? You could do this, but we lose the part of the model that indicates that a discount has been applied e.g. `discounted = undefined`. You could check that `original === final` repeatedly - but it felt as though `discountedPrice === undefined` was more explicit.

Part of this is trying to remove as many inferences from the data as I can, while allowing you to render the different types of UI.